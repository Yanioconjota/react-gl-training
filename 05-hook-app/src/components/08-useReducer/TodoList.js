import React from 'react';
import TodoListItem from './TodoListItem';
import PropTypes from "prop-types";

const TodoList = ({todos, handleDelete, handleToggle}) => {
  return (
    <ol className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          index={i}
          handleDelete={handleDelete}
          handleToggle={handleToggle}
        />
      ))}
    </ol>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleDelete: PropTypes.func,
  handleToggle: PropTypes.func
}

export default TodoList;
