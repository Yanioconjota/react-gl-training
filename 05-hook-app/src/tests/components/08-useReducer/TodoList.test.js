import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

const handleDelete = jest.fn();
const handleToggle = jest.fn();

//Las funciones no se prueban peros se pasan como referencia
const wrapper = shallow(
  <TodoList
    todos={demoTodos}
    handleDelete={handleDelete}
    handleToggle={handleToggle}/>
);

describe('pruebas con el componente TodoList', () => {
  test('Debe mostrar el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe tener 2 todoListItem', () => {
    //Compara la cantidad de TodoListItem renderizados con los items de demosTodos
    const todoListItem = wrapper.find('TodoListItem');
    expect(todoListItem.length).toBe(demoTodos.length);
    console.log(todoListItem.at(0).prop('handleDelete'));
    //Valida que el componente TodoListItem espere una funci√≥n
    expect(todoListItem.at(0).prop('handleDelete')).toEqual(expect.any(Function));
    expect(todoListItem.at(0).prop('handleToggle')).toEqual(expect.any(Function));
  });
  
  
});
