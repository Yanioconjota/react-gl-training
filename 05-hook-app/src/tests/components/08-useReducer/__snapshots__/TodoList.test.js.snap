// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con el componente TodoList Debe mostrar el componente 1`] = `
<ol
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ol>
`;
